runtime/compiler/control/rossa.cpp

```diff
@@ -1507,13 +1507,18 @@ onLoadInternal(
    else
       compInfo->getCpuUtil()->updateCpuUtil(jitConfig);
 
-#ifdef TR_TARGET_S390
    // Need to let VM know that we will be using a machines vector facility (so it can save/restore preserved regs),
    // early in JIT startup to prevent subtle FP bugs
+#ifdef TR_TARGET_S390
    if (TR::Compiler->target.cpu.supportsFeature(OMR_FEATURE_S390_VECTOR_FACILITY) && !TR::Options::getCmdLineOptions()->getOption(TR_DisableSIMD))
       {
       javaVM->extendedRuntimeFlags |= J9_EXTENDED_RUNTIME_USE_VECTOR_REGISTERS;
       }
+#elif TR_TARGET_POWER
```

Comment:
VSR is only supported in 64-bit mode, so this ifdef needs to check that (or some other check needs to be added).

Reply:
hardware-wise and JIT-wise, vector support is available in 32bit too.  maybe from reality perspective, no 32bit build beyond java8. do you mean that?

Reply:
It was a somewhat arbitrary decision - the VM helpers only do the VSR work on 64-bit, so we either need to fix that, or the JIT can't use them in 32-bit mode.

Does autoSIMD support 32-bit?

Reply:
> Does autoSIMD support 32-bit?

yes, i believed so.

Reply:
@gita-omr Do we want support on 32-bit (bearing mind the continued existence of the 32-bit IBM builds)?

Reply:
Yes, we never treated 32-bit any differently in autoSIMD. I don't think we want to disable it now. Sorry, I probably should've asked why we only save vector registers in 64-bit mode...

Reply:
I will address this in a new PR shortly.

Reply:
We've opted to defer 32-bit support, but it doesn't matter to the VM if the bit is set in 32-bit mode (all of the VM code is ifdeffed on 64-bit). The autosimd code will need to be aware that helpers only preserve VSRs on 64-bit.